client/src/controllers/ProgressController.ts

```diff
@@ -72,6 +72,14 @@ export class ProgressController extends Controller<HTMLButtonElement> {
     });
   }
 
+  deactivate() {
+    this.loadingValue = false;
+    if (this.timer) {
+      clearTimeout(this.timer);
+      this.timer = undefined;
+    }
```

Comment:
I wonder whether it would be better to move these lines to the end of `loadingValueChanged` when the value is changed to `false`?

Then, in `disconnect()`, we change the code to just call `this.deactivate()` instead of just clearing the timer.

Reply:
I like the approach, but it doesn't work, because `loadingValueChanged` is not getting called. I'm guessing that change callbacks don't get called when a controller is disconnecting.

I've removed the `this.timer = undefined;` line because it's unnecessary.